src/lib_json/json_reader.cpp

```diff
@@ -1873,20 +1868,36 @@ std::vector<OurReader::StructuredError> OurReader::getStructuredErrors() const {
 }
 
 class OurCharReader : public CharReader {
+public:
   bool const collectComments_;
   OurReader reader_;
 
-public:
   OurCharReader(bool collectComments, OurFeatures const& features)
-      : collectComments_(collectComments), reader_(features) {}
-  bool parse(char const* beginDoc, char const* endDoc, Value* root,
-             String* errs) override {
-    bool ok = reader_.parse(beginDoc, endDoc, *root, collectComments_);
-    if (errs) {
-      *errs = reader_.getFormattedErrorMessages();
+      : collectComments_(collectComments), reader_(features) {
+        this->_impl = std::unique_ptr<OurCharReader::Impl>(new OurCharReader::Impl(collectComments, features));
```

Comment:
If you're in the OurCharReader class scope you don't need to say OurCharReader here.
You can initialize _impl just like you initialized the other members, in the initialization list.

Reply:
You cannot initialize base class members from the initialization list.